File: MonkeyJump/ViewControllers/GameMenuViewController.m
//
//  GameMenuViewController.m
//  MonkeyJump
//
//  Created by Kauserali on 17/11/13.
//  Copyright (c) 2013 Raywenderlich. All rights reserved.
//

#import "GameMenuViewController.h"
#import "SKTAudio.h"

@implementation GameMenuViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self.navigationController setNavigationBarHidden:YES];
    
    if (![[SKTAudio sharedInstance] isBackgroundMusicPlaying]) {
        [[SKTAudio sharedInstance] playBackgroundMusic:@"background_track.aiff"];
    }
}

- (IBAction)newGameButtonPressed:(id)sender
{
    [self.navigationController pushViewController:[self.storyboard instantiateViewControllerWithIdentifier:@"GameViewController"] animated:YES];
}

- (IBAction)gameCenterButtonPressed:(id)sender
{
    NSLog(@"Game center button pressed");
}
@end

File: MonkeyJump/ViewControllers/GameMenuViewController.swift
//
//  GameMenuViewController.swift
//  MonkeyJump
//
//  Created by philippe eggel on 21/12/2015.
//  Copyright © 2015 PhilEagleDev.com. All rights reserved.
//

import GameKit

class GameMenuViewController: UIViewController {

    
    // MARK: - ViewController Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()

        navigationController?.navigationBarHidden = true
        
        if !MonkeyAudio.sharedInstance.backgroundMusicPlaying {
            MonkeyAudio.sharedInstance.playBackgroundMusic("background_track.aiff")
        }
    }

    
    // MARK: - User IBAction
    @IBAction func newGameButtonPressed(sender: UIButton) {
        navigationController?.pushViewController(storyboard!.instantiateViewControllerWithIdentifier("GameViewController"), animated: true)
    }

    @IBAction func gameCenterButtonPressed(sender: UIButton) {
        GameKitHelper.sharedInstance.presentGameCenterViewControllerFromViewController(self)
    }
    
    @IBAction func challengesReceivedButtonPressed(sender: UIButton) {
        let challengeViewController = ChallengesPickerViewController(nibName: "ChallengesPickerViewController", bundle: nil)
        
        challengeViewController.challengeCanceledBlock = {
            self.dismissViewControllerAnimated(true, completion: nil)
        }
        
        challengeViewController.challengeSelectedBlock = { (challenge: GKScoreChallenge) -> Void in
            self.dismissViewControllerAnimated(true, completion: nil)
            
            let gameViewController = self.storyboard?.instantiateViewControllerWithIdentifier(GameViewController.storyboardID) as! GameViewController
            if challenge.score!.context != 0 {
                HttpClient.sharedInstance.getGameTrackRecordDetailsForKey(challenge.score!.context, completionHandler: { (response, error) -> () in
                    if error == nil {
                        gameViewController.gameTrackRecord = response
                        self.navigationController?.pushViewController(gameViewController, animated: true)
                    }
                    else {
                        self.navigationController?.pushViewController(gameViewController, animated: true)
                    }
                })
                
            } else {
                self.navigationController?.pushViewController(gameViewController, animated: true)
            }
        }
        
        let nc = UINavigationController(rootViewController: challengeViewController)
        presentViewController(nc, animated: true, completion: nil)
    }
}
